script.js

```diff
@@ -1,4 +1,704 @@
+    isBlackJack = true;
+  }
+  // except that scenario, it would be all NOT black jack and can return directly "false"
+  return isBlackJack;
+};
+
+var calculateTotalHandValue = function (handArray) {
+  var totalHandValue = 0;
+  var aceCounter = 0;
+
+  for (var i = 0; i < handArray.length; i += 1) {
+    var currentCard = handArray[i];
+    // console.log(currentCard);
+    // Jack / Queen / King equals to 10
+    if (
+      currentCard.name == "jack" ||
+      currentCard.name == "queen" ||
+      currentCard.name == "king"
+    ) {
+      totalHandValue = totalHandValue + 10;
+    } else if (currentCard.name == "ace") {
+      totalHandValue = totalHandValue + 11;
+      aceCounter += 1;
+    }
+    // if not, just face value
+    else {
+      totalHandValue = totalHandValue + currentCard.rank;
+    }
+    for (var j = 0; j < aceCounter; j += 1) {
+      if (totalHandValue > 21) {
+        totalHandValue = totalHandValue - 10;
+      }
+    }
+  }
+  return totalHandValue;
+};
+
+var displayPlayerAndDealerHandsFirst = function (
+  playerHandArray,
+  dealerHandArray
+) {
+  //playerHand
+  var playerMessage = "Player Hand : <br>";
+  for (var index = 0; index < playerHandArray.length; index += 1) {
+    playerMessage =
+      playerMessage +
+      "- " +
+      playerHandArray[index].name +
+      " of " +
+      playerHandArray[index].suit +
+      "<br>";
+  }
+  //dealerHand
+  var dealerMessage =
+    "Dealer Hand : <br>" +
+    "- " +
+    dealerHandArray[0].name +
+    " of " +
+    dealerHandArray[0].suit +
+    "<br>- Unknown Card";
+  return playerMessage + "<br>" + dealerMessage + "<br><br>";
+};
+
+var displayPlayerAndDealerHands = function (playerHandArray, dealerHandArray) {
+  //playerHand
+  var playerMessage = "Player Hand : <br>";
+  for (var index = 0; index < playerHandArray.length; index += 1) {
+    playerMessage =
+      playerMessage +
+      "- " +
+      playerHandArray[index].name +
+      " of " +
+      playerHandArray[index].suit +
+      "<br>";
+  }
+  //dealerHand
+  var dealerMessage = "Dealer Hand : <br>";
+  for (var index = 0; index < dealerHandArray.length; index += 1) {
+    dealerMessage =
+      dealerMessage +
+      "- " +
+      dealerHandArray[index].name +
+      " of " +
+      dealerHandArray[index].suit +
+      "<br>";
+  }
+  return playerMessage + "<br>" + dealerMessage + "<br><br>";
+};
+
+var displayTotalValueFirst = function (playerHandValue, dealerHandArray) {
+  var totalHandValueMessage =
+    "<br><br><br>Player Total Hand Value : " +
+    playerHandValue +
+    "<br>Dealer Total Hand Value : " +
+    dealerHandArray[0].rank +
+    " + 'unknown card' ";
+  return totalHandValueMessage;
+};
+
+var displayTotalValue = function (playerHandValue, dealerHandValue) {
+  var totalHandValueMessage =
+    "<br><br><br>Player Total Hand Value : " +
+    playerHandValue +
+    "<br>Dealer Total Hand Value : " +
+    dealerHandValue;
+  return totalHandValueMessage;
+};
+
+var deck = [
```

Comment:
You could have used the deck function given in the course material or even better, use a for loop to create a deck. You almost never want to hard code things like these